src/main/java/io/github/dsheirer/module/decode/DecoderFactory.java

```diff
@@ -226,15 +226,19 @@ private static void processMPT1327(ChannelMapModel channelMapModel, Channel chan
         ChannelMap channelMap = channelMapModel.getChannelMap(mptConfig.getChannelMapName());
         Sync sync = mptConfig.getSync();
         modules.add(new MPT1327Decoder(sync));
-        modules.add(new AudioModule(aliasList));
+
+        final int timeout = mptConfig.getCallTimeout() * 1000;
+        // Set max segment audio sample length slightly above call timeout to
+        // not create a new segment if the processing chain finishes a bit after
+        // actual call timeout.
+        modules.add(new AudioModule(aliasList, timeout + 5000));
```

Comment:
My intent here, based on your input in the issue description, was to not create next segment in case processing chain will be slightly late with stopping traffic channel processing, mostly for the purposes of recording. But I'm not entirely sure that's the right way to go.

Reply:
I think the 5 second buffer should be good for handling this case.